src/Digdir.Domain.Dialogporten.GraphQL/Common/Authorization/AuthorizationOptionsSetup.cs

```diff
@@ -41,5 +47,26 @@ public void Configure(AuthorizationOptions options)
         options.AddPolicy(AuthorizationPolicy.Testing, builder => builder
             .Combine(options.DefaultPolicy)
             .RequireScope(AuthorizationScope.Testing));
+
+        options.AddPolicy(AuthorizationPolicy.EndUserSubscription, policy => policy
+            .Combine(options.GetPolicy(AuthorizationPolicy.EndUser)
+                     ?? throw new InvalidOperationException($"{AuthorizationPolicy.EndUser} policy not found"))
+            .RequireAssertion(context =>
+            {
+                if (context.Resource is not AuthorizationContext authContext) return false;
+
+                if (authContext.Document.Definitions.Count == 0) return false;
+
+                var definition = authContext.Document.Definitions[0];
+
+                if (definition is not OperationDefinitionNode operationDefinition) return false;
+
+                if (operationDefinition.Operation != OperationType.Subscription) return false;
+
+                if (!operationDefinition.TryGetDialogEventsSubscriptionDialogId(out var dialogId)) return false;
```

Comment:
Denne lambdaen er litt busy. Ser du har en extension metode på OperationDefinitionNode. Kunne dette heller vært en extension på AuthorizationHandlerContext (med samme navn) slik at alt dette veldig spesielle uthentingen av dialogid var litt skjult for auth konteksten? Resultatet kunne blitt noe slikt: 

```csharp
.RequireAssertion(context =>
    context.TryGetDialogEventsSubscriptionDialogId(out var dialogIdTopic)
     && context.User.TryGetClaimValue(DialogTokenClaimTypes.DialogId, out var dialogIdClaimValue)
     && Guid.TryParse(dialogIdClaimValue, out var dialogIdClaim)
     && dialogIdTopic == dialogIdClaim)
```